docs/asciidoc/static/deploying.asciidoc

```diff
@@ -0,0 +1,145 @@
+
+// [float]
+[[deploying-message-queueing]]
+=== Managing Throughput Spikes with Message Queueing
+
+When a Logstash pipeline produces data faster than the Elasticsearch cluster can ingest, use a message queue to serve 
+as a buffer. By default, Logstash propagates backpressure to throttle incoming events when indexer consumption rates 
+fall below incoming data rates. Since this throttling can lead to events being buffered at the data source, preventing 
+backpressure with message queues becomes an important part of managing your deployment.
+
+Adding a message queue to your Logstash deployment also provides a level of protection from data loss. When a Logstash 
+instance that has consumed data from the message queue fails, the data can be replayed from the message queue to an 
+active Logstash instance.
+
+Several third-party message queues exist, such as Redis, Kafka, or RabbitMQ. Logstash provides input and output plugins 
+to integrate with several of these third-party message queues. When your Logstash deployment has a message queue 
+configured, Logstash functionally exists in two phases: shipping instances, which handles data ingestion and storage in 
+the message queue, and indexing instances, which retrieve the data from the message queue, apply any configured 
+filtering, and write the filtered data to an Elasticsearch index.
+
+image::images/deploy_5.png[]
+
+// [float]
+[[deploying-logstash-ha]]
+=== Multiple Connections for Logstash High Availability
+
+To make your Logstash setup more resilient to individual instance failures, you can set up a load balancer between your 
+data source machines and the Logstash cluster. The load balancer handles the individual connections to the Logstash 
+instances, ensuring that when an individual instance becomes unavailable, data ingestion and processing can continue.
+
+image::images/deploy_6.png[]
+
+The previous diagram is unable to process input from a specific type, such as an RSS feed or a file, if the Logstash 
+instance dedicated to that input type becomes unavailable. For more robust input processing, configure each Logstash 
+instance for multiple inputs, as in the following diagram:
+
+image::images/deploy_7.png[]
+
+// [float]
+[[deploying-scaling]]
+=== Scaling Logstash
+
+A mature Logstash deployment typically has the following pipeline:
+
+* The _input_ section consumes data from the source, and consists of Logstash instances with the proper input plugins.
+* The _message queue_ serves as a buffer to hold ingested data and serve as failover protection.
+* The _filter_ section applies parsing and other processing to the data consumed from the message queue.
+* The _indexing_ section moves the processed data into Elasticsearch.
+
+Any of these layers can be scaled by adding computing resources. Examine the performance of these components regularly 
+as your use case evolves and add resources as needed. The presence of input throttling or backpressure can indicate the 
```

Comment:
"If incoming events are being throttled, you might need more storage for the message queue. Alternatively, you can increase the rate of data consumption by adding Logstash instances."?